packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -626,6 +597,72 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		}
 	}
 
+	/**
+	 * Creates a {@link RevertibleAlpha} object that can undo a specific change in the tree's history.
+	 *
+	 * @param revision - The revision tag identifying the change to be made revertible.
+	 * @param kind - The kind of commit (e.g., Default, Undo, Redo) this revertible represents.
+	 * @param checkout - The {@link TreeCheckout} instance this revertible belongs to.
+	 * @param onRevertibleDisposed - Callback function that will be called when the revertible is disposed.
+	 * @returns - {@link RevertibleAlpha}
+	 */
+	private createRevertible(
+		revision: RevisionTag,
+		kind: CommitKind,
+		checkout: TreeCheckout,
+		onRevertibleDisposed: ((revertible: RevertibleAlpha) => void) | undefined,
+	): RevertibleAlpha {
+		const commitBranches = checkout.revertibleCommitBranches;
+
+		const revertible: RevertibleAlpha = {
+			get status(): RevertibleStatus {
+				const revertibleCommit = commitBranches.get(revision);
+				return revertibleCommit === undefined
+					? RevertibleStatus.Disposed
+					: RevertibleStatus.Valid;
+			},
+			revert: (release: boolean = true) => {
+				if (revertible.status === RevertibleStatus.Disposed) {
+					throw new UsageError("Unable to revert a revertible that has been disposed.");
+				}
+
+				const revertMetrics = checkout.revertRevertible(revision, kind);
+				checkout.logger?.sendTelemetryEvent({
+					eventName: TreeCheckout.revertTelemetryEventName,
+					...revertMetrics,
+				});
+
+				if (release) {
+					revertible.dispose();
+				}
+			},
+			clone: (forkedBranch?: TreeBranch) => {
```

Comment:
Cloning a revertible onto the same branch is useful for an application that wants to provide ownership of a revertible to multiple components. Suppose I have two components in my application that:
1. Don't know about each other
2. Are operating on the same branch (e.g. the main branch)
3. Both want to be able to revert (e.g. they both have undo buttons)

You'd want to clone the revertible, giving one to the first component and the clone to the second. Or, perhaps the parent component retains ownership of the original revertible and gives a new clone to each subcomponent. If you can't clone the revertible for the same branch, then you have to share the same revertible between the parent and both components, and now they _do_ need to know about each other so that they know when to (or when not to) dispose the revertible.

Regarding the parameter being optional, it's just a convenience. For an application like the above, which operates solely on the main branch, it would be annoying at best and confusing at worst to be forced to pass in the main branch every time it wants to do a clone. With it being optional, the user doesn't need to understand the branching feature at all in order to clone revertibles.

Reply:
I agree that it would be useful, but I'm not entirely convinced that the changes here sufficiently support it. At the very least, I think it needs to be documented better. 

some thoughts:
- if one of the components does a revert, how does it get communicated to the other? do we just fail whenever the other ones tries to do a revert and we can't find the repair data anymore?
- it seems like the status is currently cloned as well, which is fine since revertibles can be disposed without being reverted, but I think we'd need some way to properly communicate between clones whether the repair data they share can still be used
- do we maybe want a different api for cloning with the same branch? it kinda sounds like the meaning changes depending on if we're cloning with a different branch or the same branch i.e. with a different branch, nothing you do with the cloned revertible will affect the original but with the same branch, they still have the same dependencies even though they can be owned by different components and disposed separately as well although i guess it would be somewhat similar to the case where a user explicitly does not dispose after revert and we call revert again on the same revertible and it just noops. which I still think it kinda strange.
- I think having the parameter be optional is fine if the behavior around operating on the same branch is more clear

Reply:
For a set of clones of the same revertible associated with the same branch, the repair data will not be collected until _all_ clones have called `dispose()`. This is the intended behavior, and I believe is the currently behavior, but if it's not covered by tests then we should add more tests.

So, yes, if component A and component B each have a clone of a revertible, and are on the same branch, then when A does a revert it will do something, and when B does a revert it will (probably) no-op. This is what I would expect - the reason you cloned in the first place is so that A and B don't have to know what each other are doing, therefore, if B does an undo but it was already undone by A, nothing will happen (because it "already happened"). The other option which you alluded to in your first bullet is for B's undo stack to be changed along with A's undo stack. But if that's what the user wants, then A and B are not isolated and the user shouldn't have done a clone in the first place. Undo-stacks are probably 1:1 with branches most of the time, is my guess. Being able to clone a revertible on the same branch is more of a ref-counting thing. Like you have some code in your application that wants to ensure it can revert some specific data at some point in the future, so it holds on to a particular revertible - regardless of whether or not a clone of that revertible might get reverted anyway by some other means. I think it's likely to be a pretty uncommon scenario, but it might be useful to have as an option, it's easy for us to implement, and it's the natural behavior (IMO) of a parameterless call to `clone`.

A review of potential scenarios (e.g. with Nick) to inform the documentation would probably not be a bad idea. If we find that it's just too hard to explain why you would do this, then that could be a sign that we should just leave it out.

Reply:
I don't think this is the current behavior at all. Neither TreeCheckout.disposeRevertible nor the dispose function on the created Revertible makes any kind of checks and we don't do any ref counting. Before this PR, it wouldn't have mattered because we only allowed a single revertible to be obtained for a revision by the getRevertible function so it wasn't being tested and I don't think this pr adds any coverage for that scenario, unless I'm missing something?

I think the reasons you listed make sense and it could be useful, but I don't think it should be part of this PR. It seems like it would require a lot more thought and testing.

Reply:
I think you're right - looking at the code I'm guessing that the maps pertaining to revertibles will get clobbered when the first revertible for a given branch is disposed.

Reply:
Ji, I agree you should just remove this behavior for now. You did write a test for it, but the test is wrong. The test asserts that for two revertibles associated with the same branch, once one is disposed, the other will error when it tries to revert. That defeats the whole point of cloning in the first place (which is to create a unique, isolated owned reference).